### PR TITLE
Allow substitution of profile values and properties using `${}` with …

### LINKED ISSUE #14977: Allow substitution of variables in connection profiles with preferences
Allow values in connection profiles to contain `${…}` and replace when reading with value from preferences.

### PR DESCRIPTION
**Pull request by dkocher**
…value from preferences. Resolves #14977.

### CODE REVIEW

File: core/src/main/java/ch/cyberduck/core/Profile.java [hunk 1]
```diff
@@ -570,15 +576,17 @@ public boolean isUTCTimezone() {
...
         if(null == list) {
             return Collections.emptyList();
         }
-        return list;
+        final ArrayList<String> substituted = new ArrayList<>(list);
+        substituted.replaceAll(substitutor::replace);
+        return substituted;
     }
 
     private boolean bool(final String key) {
```
Comment (chenkins) [inline core/src/main/java/ch/cyberduck/core/Profile.java:592]: no support for injecting `bool`s?

File: core/src/main/java/ch/cyberduck/core/Profile.java [hunk 2]
```diff
@@ -29,11 +29,14 @@
 import org.apache.commons.codec.binary.Base64;
```
Comment (chenkins) [inline core/src/main/java/ch/cyberduck/core/Profile.java:29]: No unit tests in this mr?

Comment (chenkins) [inline core/src/main/java/ch/cyberduck/core/Profile.java:29]: No unit tests in this mr?

File: core/src/main/java/ch/cyberduck/core/Profile.java (hunk 1 above)
Comment (dkocher) [reply core/src/main/java/ch/cyberduck/core/Profile.java:592]: Not possible as the property list dtd does not allow arbitrary values in this case.

**Review by AliveDevil — APPROVED** [review]

**Pull request merged**